### DISCUSSION TITLE
pleas add a 3 dimensional  mode virtual reality mode and augmented reality mode for an addon.

### DISCUSSION THREAD

**Discussion by zacplatformer** [ScratchAddons/ScratchAddons#5052] [Ideas]
i think modes would work verry well, like you would select, 2d, 3d, vr, ar or coustom modes. coustom, you would be able to combine 2d, 3d, vr, or ar.  Just a thought. maybe a playstation/xbox mode. those two wont work mainly because ps# or xbox# do not have web browser.

**Comment by BroJac5246** [comment]
Umm... sorry to tell you this, but it's pretty hard to implement those sorts of things, plus that's pretty far out of the scope of Scratch Addons. Scratch Addons doesn't work magic- it just improves what Scratch has created.

Also, how would Scratch even run VR projects? It's not powerful enough.

**Reply by zacplatformer** [reply]
yhea. just 3d blocks would work. 3d editor. chose between 3d and 2d

**Reply by BroJac5246** [reply]
That wouldn't work for people who don't have Scratch Addons. No Scratch Addons feature should negatively impact the experience of others.
